datafusion/core/src/dataframe.rs

```diff
@@ -627,22 +627,12 @@ impl DataFrame {
     /// # }
     /// ```
     pub async fn count(self) -> Result<usize> {
-        let rows = self
-            .aggregate(
-                vec![],
-                vec![datafusion_expr::count(Expr::Literal(ScalarValue::Null))],
```

Comment:
This was working by accident, as the null count for NullArray was computed incorrectly